### ISSUE TITLE
sqlite table options combo not supported (strict & without rowid)

### ISSUE THREAD

**Issue by dcecile** [sqlfluff/sqlfluff#5452]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Tried to lint SQL that creates a strict table without a row ID.

```sql
create table version (id integer primary key) strict without rowid;
```

### Expected Behaviour

Linting works OK

### Observed Behaviour

Got an "unparsable section" error

```
== [script.sql] FAIL                                                                                         
L:   1 | P:  54 |  PRS | Line 1, Position 54: Found unparsable section: 'without
                       | rowid;'
WARNING: Parsing errors found and dialect is set to 'sqlite'. Have you configured your dialect correctly?
All Finished 📜 🎉!
```

### How to reproduce

```sh
echo "create table version (id integer primary key) strict without rowid;" >script.sql
sqlfluff lint script.sql --dialect=sqlite
```

The `without rowid` syntax was added for https://github.com/sqlfluff/sqlfluff/issues/1528. And `strict` is also supported. But they can't be used together, regardless of order.

According to the SQLite docs, both are allowed, in any order: https://www.sqlite.org/syntax/table-options.html

### Dialect

SQLite

### Version

```
sqlfluff, version 2.3.5
Python 3.11.6
```

### Configuration

No config

### Are you willing to work on and submit a PR to address the issue?

- [x] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by dcecile** [comment]
Sorry! I was missing the `,` as specified in the docs.

Fixed repro passes :heavy_check_mark: 

```sh
echo "create table version (id integer primary key) strict, without rowid;" >script.sql
sqlfluff lint script.sql --dialect=sqlite
```
